packages/react-storage/src/components/StorageBrowser/composables/Search.tsx

```diff
@@ -17,19 +10,22 @@ const BLOCK_NAME = `${CLASS_BASE}__search`;
 const TOGGLE_BLOCK = 'toggle';
 
 export interface SearchProps {
-  onSearch?: (term: string, includeSubfolders: boolean) => void;
+  onSearch?: () => void;
+  onSearchClear?: () => void;
+  searchQuery?: string;
+  onSearchQueryChange?: (query: string) => void;
   searchPlaceholder?: string;
-  showIncludeSubfolders?: boolean;
+  children?: React.ReactNode;
 }
```

Comment:
Since you are touching the structure of the component code could you make some updates, can be done in a follow up:
```tsx
export interface SearchFieldProps {
  id?: string;
  label?: React.ReactNode
  query?: string;
  onQuery?: (query: string) => void;
  onSearch?: () => void;
  onClear?: () => void;
  placeholder?: string;
}
```